File: src/Deopeia.Finance.Web/src/models/trading/instrument.ts
import type { Session } from './session';

export interface Instrument {
  symbol: string;
  name: string;
  currencyCode: string;
  pricePrecision: number;
  tickSize: number;
  contractSize: ContractSize;
  volumeRestriction: VolumeRestriction;
  sessions: Session[];
  leverages: number[];
}

export interface ContractSize {
  quantity: number;
  unitCode: string;
}

export interface VolumeRestriction {
  min: number;
  max: number;
  step: number;
}

export interface InstrumentMap {
  [key: string]: Instrument;
}

File: src/Deopeia.Finance.Bff/Models/Instruments/Instrument.cs
namespace Deopeia.Finance.Bff.Models.Instruments;

public class Instrument
{
    public string Symbol { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public UnderlyingType UnderlyingType { get; set; }

    public string CurrencyCode { get; set; } = string.Empty;

    public decimal PricePrecision { get; set; }

    public decimal TickSize { get; set; }

    public ContractSize ContractSize { get; set; } = new();

    public VolumeRestriction VolumeRestriction { get; set; } = new();

    public ICollection<decimal> Leverages { get; set; } = [];
}
